tools/crm_resource_print.c

```diff
@@ -152,6 +152,57 @@ attribute_list_default(pcmk__output_t *out, va_list args) {
     return pcmk_rc_ok;
 }
 
+PCMK__OUTPUT_ARGS("agent-status", "int", "const char *", "const char *", "const char *",
+                  "const char *", "const char *", "int")
+static int
+agent_status_default(pcmk__output_t *out, va_list args) {
+    int status = va_arg(args, int);
+    const char *action = va_arg(args, const char *);
+    const char *name = va_arg(args, const char *);
+    const char *class = va_arg(args, const char *);
+    const char *provider = va_arg(args, const char *);
+    const char *type = va_arg(args, const char *);
+    int rc = va_arg(args, int);
+
+    if (status == PCMK_LRM_OP_DONE) {
+        out->info(out, "Operation %s%s%s (%s:%s:%s) returned: '%s' (%d)",
+                  action, name ? " for " : "", name ? name : "", class,
+                  provider ? provider : "", type,
```

Comment:
The second : should be printed only if provider is given (example in crm_generate_ra_key(), we don't do that everywhere in log messages but we really should)

I could see maybe having some macros PCMK__RA_FMT and PCMK__RA_FMT_ARGS(c,p,t) to simplify it a bit

Reply:
Fixed.  Yeah, some macros would be nice but that seems liek a job for another PR.